app/controllers/concerns/signature_verification.rb

```diff
@@ -123,13 +136,36 @@ def build_signed_string(signed_headers)
   end
 
   def matches_time_window?
-    begin
-      time_sent = Time.httpdate(request.headers['Date'])
-    rescue ArgumentError
-      return false
+    created_time = nil
+    expires_time = nil
+
+    if signature_algorithm == 'hs2019' && signature_params['created'].present?
```

Comment:
Can we enforce `created` is present for users of the `hs2019` algorithm? That would solve one of the main security issues I've previously had with HTTP-Signatures (Servers can shoot themselves in the foot by not specifying a `created` time)

Reply:
Spec also says we shouldn't verify the signature if `(created)` is part of the `signed_headers` but `created` isn't present in the params but —do we do that?

Reply:
We could enforce the presence of `created`. It's not mandated by the spec, but we are allowed to require more than the spec requires.

As for checking the arguments are here… my code doesn't explicitly fails, but since the arguments are included in the signature input, it will fail anyway. I will change it to fail with an explicit error, though.